Modules/Extensions/CollectionExtensions.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TOHE
+{
+    public static class CollectionExtensions
+    {
+        /// <summary>
+        /// Returns a random element from a collection
+        /// </summary>
+        /// <param name="collection">The collection</param>
+        /// <typeparam name="T">The type of the collection</typeparam>
+        /// <returns>A random element from the collection, or the default value of <typeparamref name="T"/> if the collection is empty</returns>
+        public static T RandomElement<T>(this IList<T> collection)
+        {
+            if (collection.Count == 0) return default;
+            return collection[IRandom.Instance.Next(collection.Count)];
+        }
+        /// <summary>
+        /// Shuffles all elements in a collection randomly
+        /// </summary>
+        /// <typeparam name="T">The type of the collection</typeparam>
+        /// <param name="collection">The collection to be shuffled</param>
+        /// <returns>The shuffled collection as a <see cref="List{T}"/></returns>
+        public static List<T> Shuffle<T>(this IEnumerable<T> collection)
+        {
+            var list = collection.ToList();
+            int n = list.Count;
+            var r = IRandom.Instance;
+            while (n > 1)
+            {
+                n--;
+                int k = r.Next(n + 1);
+                (list[n], list[k]) = (list[k], list[n]);
+            }
+            return list;
+        }
```

Comment:
This already exists in the code

![image](https://github.com/0xDrMoe/TownofHost-Enhanced/assets/104814436/fd7165e7-576d-431a-810f-0add08377d40)